impala/hiveserver2.py

```diff
@@ -1003,18 +1003,19 @@ def __init__(self, client, retries=3):
         self.client = client
         self.retries = retries
 
-    def _rpc(self, func_name, request):
+    def _rpc(self, func_name, request, retry_on_http_error=False):
         self._log_request(func_name, request)
-        response = self._execute(func_name, request)
+        response = self._execute(func_name, request, retry_on_http_error)
         self._log_response(func_name, response)
         err_if_rpc_not_ok(response)
         return response
 
-    def _execute(self, func_name, request):
+    def _execute(self, func_name, request, retry_on_http_error=False):
         # pylint: disable=protected-access
         # get the thrift transport
         transport = self.client._iprot.trans
         tries_left = self.retries
+        last_http_exception = None
         while tries_left > 0:
```

Comment:
Not this change, but I think this condition should be "tries_left >= 0" instead.
(I assume self.retries == 0 means the call should be attempted once but not retried after a failure).

Reply:
I think self.retries == 1 means that the call should be attempted once without retries
I agree that you could argue that this could be self.tries as when it is 1 there is no actual 'retry'.
I am going to leave the naming like this for now.
But I am making the change you suggest on L1041, thank you.